AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift

```diff
@@ -41,3 +42,22 @@ public class AmplifyAWSServiceConfiguration: AWSServiceConfiguration {
         super.init(region: regionType, credentialsProvider: nil)
     }
 }
+
+extension AmplifyAWSServiceConfiguration {
+
+    static var platformMapping: [Platform: String] = [:]
+
+    public static func addUserAgentPlatform(_ platform: Platform, version: String) {
+        platformMapping[platform] = version
+    }
+
+    public enum Platform: String {
+        case flutter = "amplify-flutter"
+    }
+
+    static func platformInformation() -> String {
+        let sdkName = "amplify-iOS"
+        let additionalPlatform = platformMapping.map {"\($0.rawValue)/\($1) "}.joined()
```

Comment:
Let's use `joined` all the way down; it'll be less brittle than appending the space at the end of every token:

```swift
var platformTokens = platformMapping.map { "\($0.rawValue)/\($1)" }
platformTokens.append("amplify-iOS/\(version)")
let additionalPlatforms = platformTokens.joined(separator: " ")
```

Reply:
Refactored